src/pkg/docs/service/composition/composition.go

```diff
@@ -0,0 +1,126 @@
+package composition
+
+import (
+	"encoding/json"
+	"github.com/google/uuid"
+	"hms/gateway/pkg/config"
+	"hms/gateway/pkg/crypto/chacha_poly"
+	"hms/gateway/pkg/docs/model"
+	"hms/gateway/pkg/docs/model/base"
+	"hms/gateway/pkg/docs/service"
+	"hms/gateway/pkg/docs/types"
+	"log"
+	"time"
+)
+
+type CompositionService struct {
+	Doc *service.DefaultDocumentService
+	Cfg *config.Config
+}
+
+func NewCompositionService(docService *service.DefaultDocumentService, cfg *config.Config) *CompositionService {
+	return &CompositionService{
+		Doc: docService,
+		Cfg: cfg,
+	}
+}
+
+func (s CompositionService) ParseJson(data []byte) (*model.Composition, error) {
+	var doc model.Composition
+	err := json.Unmarshal(data, &doc)
+	if err != nil {
+		return nil, err
+	}
+	return &doc, nil
+}
+
+func (s CompositionService) MarshalJson(doc *model.Composition) ([]byte, error) {
+	return json.Marshal(doc)
+}
+
+func (s CompositionService) Create(userId string, request *model.CompositionCreateRequest) (*model.Composition, error) {
+	return s.CreateWithId(userId, uuid.New().String(), request) // TODO no new???
```

Comment:
The new Composition will come directly in the body of the request

Reply:
\+